pkg/endpoint/endpoint.go

```diff
@@ -1181,24 +1181,23 @@ func (e *Endpoint) Update(owner Owner, opts models.ConfigurationMap) error {
 
 	// Option changes may be overridden by the policy configuration.
 	// Currently we return all-OK even in that case.
-	changed, ctCleaned, err := e.TriggerPolicyUpdatesLocked(owner, opts)
+	_, ctCleaned, err := e.TriggerPolicyUpdatesLocked(owner, opts)
 	if err != nil {
 		e.Mutex.Unlock()
 		ctCleaned.Wait()
 		return UpdateCompilationError{err.Error()}
 	}
 
-	if changed {
-		changed = e.SetStateLocked(StateWaitingToRegenerate, "Updated endpoint options; policy changes apply to this endpoint")
-	}
-	e.Mutex.Unlock()
-	ctCleaned.Wait()
-
 	reason := "endpoint was updated via API"
 	if opts == nil {
 		reason = "endpoint was manually regenerated via API"
 	}
 
+	_ = e.SetStateLocked(StateWaitingToRegenerate, reason)
```

Comment:
@jrajahalme is it OK to ignore the result here? The desired behavior in the commit that necessitated this patch says ```// Update modifies the endpoint options and *always* regenerates the program.``` 

I think we should only regenerate if the state transition succeeds, otherwise we might have endpoints in not-ready state. However, this contradicts the new behavior of this function as previously specified.

Reply:
No, I looked into this and this is why I haven't submitted it yet. We need to make the building dependant on `SetStateLocked()` returning true.

That said, I'll be doing a follow-up that allows to overwrite this and to a full identity resolution, policy calc and rebuild manually so we can unstuck stuck endpoints.

Reply:
Otherwise users can break the state machine. This definitely needs improvements.

Reply:
OK I'll update the patch accordingly. Thanks !